flaky/_flaky_plugin.py

```diff
@@ -146,25 +147,27 @@ def _handle_test_success(self, test):
             current_passes
         )
         flaky = self._get_flaky_attributes(test)
-        min_passes = flaky[FlakyNames.MIN_PASSES]
-        self._stream.writelines([
-            ensure_unicode_string(name),
-            ' passed {0} out of the required {1} times. '.format(
-                current_passes,
-                min_passes,
-            ),
-        ])
-        if not self._has_flaky_test_succeeded(flaky):
-            self._stream.write(
-                'Running test again until it passes {0} times.\n'.format(
+        need_reruns = not self._has_flaky_test_succeeded(flaky)
+        if self._flaky_success_report:
+            min_passes = flaky[FlakyNames.MIN_PASSES]
+            self._stream.writelines([
+                ensure_unicode_string(name),
+                ' passed {0} out of the required {1} times. '.format(
```

Comment:
I find this similar message confusing:

     _failure_message = ' failed; it passed {0} out of the required {1} times.'

As it will say:

     xxx failed; it passed 0 out of the required 1 times

It reads weird because the first work ("passed") is affirmative. But the following 0 makes it a negative.  I think this would be easier to understand

     failed; test must pass X out of Y runs, but only passed Z times.

It's also a bit misleading in that if say a test needs to pass 4 out of 5 times... if it fails 2 out of the first three, the final 2 attempts will be skipped and the output will read:

     failed; it passed 1 out of the required 4 times

leaving it unsaid that it stopped at 3 attempts. So I'd amend my proposed text to include an additional message only in cases where the full number of attempts weren't used

     failed; test must pass X out of Y times, but only passed Z times with just P attempts remaining

     failed; test must pass 4 out of 5 times, but only passed 1 times [sic] with 2 attempts remaining

I actually have other thoughts. But not the time. I should just post a PR myself :)